### DISCUSSION TITLE
Problems with .update() method to change conditional access policy for SPO site using PNP.

### DISCUSSION THREAD

**Discussion by vindi4i** [pnp/powershell#220] [Q&A]
Hello,

I am trying to change conditional access policy for SharePoint Online site using PNP.
I did this previously using code:

$url = 'https://companygroup.sharepoint.com/sites/TeamBlockPolicy'
Connect-pnponline $url -useweblogin
$context = Get-PnPContext
$site = Get-PnPTenantSite -Url  $url
$site.ConditionalAccessPolicy = [Microsoft.Online.SharePoint.TenantManagement.SPOConditionalAccessPolicyType]::BlockAccess
$site.Update()
Invoke-PnPQuery

Code above not working anymore. When I call $site.Update() I get error:
'Method invocation failed because [PnP.PowerShell.Commands.Model.SPOSite] does not contain a method named 'Update'.'
Command "$site | gm" does not show method 'Update'.

However if I import legacy module SharePointPnPPowerShellOnline everything works fine. .

Could someone explain how conditional access policy for SPO site can be configured using new PNP?

PNP : 1.2.0
SharePointPnPPowerShellOnline 3.29.2101.0
PowerShell : 5.1

**Comment by YuriySamorodov** [comment]
Hey @vindi4i 
Have you considered the following command?
`Set-PnPTenantSite -ConditionalAccessPolicy BlockAccess`

**Reply by vindi4i** [reply]
Hello,

Based on the documentation (https://docs.microsoft.com/en-us/powershell/module/sharepoint-pnp/set-pnptenantsite?view=sharepoint-ps) Set-PnPTenantSite  does not have parameter '-ConditionalAccessPolicy'.

Set-SPOSite supports this parameter, however it does not work properly.

**Reply by erwinvanhunen** [reply]
The documentation is not in sync with the cmdlet itself. The Set-PnPTenantSite cmdlet does have a -ConditionalAccessPolicy parameter. We try to get the docs updated, but as we all have to do this in our free time, it can take some time.

**Reply by erwinvanhunen** [reply]
As we, in the Set-PnPTenantSite do -exactly- the same thing as Set-SPOSite (the only thing we can do is consume the APIs provided to us) and it behaves not correctly, then it seems to be an issue on the server end, of which we have no control unfortunately.

**Reply by erwinvanhunen** [reply]
From the documentation about conditional access policies I see that you require to have an Azure AD Premium P1 license. Do you have that in place?

https://docs.microsoft.com/en-us/sharepoint/control-access-from-unmanaged-devices (first note)
and
https://docs.microsoft.com/en-us/azure/active-directory/conditional-access/overview#license-requirements

**Reply by YuriySamorodov** [reply]
@erwinvanhunen Personally I have Azure AD Premium P1, however I am authenticating with an app,,,

**Reply by YuriySamorodov** [reply]
>  Set-PnPTenantSite cmdlet does have a -ConditionalAccessPolicy parameter. We try to get the docs updated, but as we all have to do this in our free time, it can take some time.

@erwinvanhunen , it looks like ConditionalAccessPolicy is available with IntelliSense for me:
![image](https://user-images.githubusercontent.com/5260172/107561088-2cecbc00-6bef-11eb-8dac-f741c2e56a5f.png)

**Reply by erikcp1975** [reply]
As of 3.29.2101.0 SharePointPnPPowerShellOnline, -conditionalaccesspolicy does not work.
In addition PNP doesn't seem to support the new Conditional Access policy, authentication-context with sensitivity label's
https://docs.microsoft.com/en-us/sharepoint/authentication-context-example

**Comment by YuriySamorodov** [comment]
#231
